autoblocks/_impl/tracer.py

```diff
@@ -238,13 +214,31 @@ async def _send_event_unsafe(
             except Exception as err:
                 log.error("Unable to evaluate events. Error: %s", err, exc_info=True)
 
-        req = await global_state.http_client().post(
-            url=INGESTION_ENDPOINT,
-            json=payload.to_json(),
-            headers=self._client_headers,
-            timeout=self._timeout_seconds,
-        )
-        req.raise_for_status()
+        try:
+            if global_state.main_thread_has_finished():
```

Comment:
I would just add a small comment about how this should only happen in the normal shutdown scenario